src/Core.Nhibernate/Entities/BaseEntity.cs

```diff
@@ -1,11 +1,8 @@
 ﻿namespace IdentityServer3.Contrib.Nhibernate.Entities
 {
-    public abstract class BaseEntity : IBaseEntity
-    {
-
-    }
+    internal abstract class BaseEntity { }
 
-    public abstract class BaseEntity<TKey> : BaseEntity, IBaseEntity<TKey>
+    public abstract class BaseEntity<TKey>
```

Comment:
It's not necessary. The BaseEntity class exists as a Class with no generic <T> associated so that it can be used as an anchor by AutoMapper config setup. Other than that it isn't used.